server/datastore/mysql/migrations/tables/20240717171504_AddGlobalStatToVulnHostCounts_test.go

```diff
@@ -15,32 +15,56 @@ func TestUp_20240717171504(t *testing.T) {
 	VALUES (?, ?, ?)
 	`
 
-	// insert team count
-	_, err := db.Exec(stmt, "CVE-2024-0717", 1, 1)
+	// Insert team 1 counts
+	_, err := db.Exec(stmt, "CVE-2024-1000", 1, 1)
+	require.NoError(t, err)
+	_, err = db.Exec(stmt, "CVE-2024-2000", 1, 10)
+	require.NoError(t, err)
+
+	// Insert team 2 count
+	_, err = db.Exec(stmt, "CVE-2024-1000", 2, 2)
+	require.NoError(t, err)
+	_, err = db.Exec(stmt, "CVE-2024-2000", 2, 20)
 	require.NoError(t, err)
 
-	// insert global count
-	_, err = db.Exec(stmt, "CVE-2024-0717", 0, 1)
+	// Insert global count
+	_, err = db.Exec(stmt, "CVE-2024-1000", 0, 4)
+	require.NoError(t, err)
+	_, err = db.Exec(stmt, "CVE-2024-2000", 0, 40)
 	require.NoError(t, err)
 
 	applyNext(t, db)
 
-	// Check that the global_stat column has 0 for team_id = 1
-	selectStmt := `
-	SELECT global_stats FROM vulnerability_host_counts
-	WHERE cve = ? and team_id = ?
+	assertHostCount := func(cve string, teamID, expectedCount int, globalStat bool) {
+		t.Helper()
+		selectStmt := `
+	SELECT host_count FROM vulnerability_host_counts
+	WHERE cve = ? and team_id = ? and global_stats = ?
 	`
-	var globalStat bool
-	err = db.QueryRow(selectStmt, "CVE-2024-0717", 1).Scan(&globalStat)
-	require.NoError(t, err)
-	require.False(t, globalStat)
 
-	// Check that the global_stat column has 1 for team_id = 0
-	err = db.QueryRow(selectStmt, "CVE-2024-0717", 0).Scan(&globalStat)
-	require.NoError(t, err)
-	require.True(t, globalStat)
+		var count int
+		err = db.QueryRow(selectStmt, cve, teamID, globalStat).Scan(&count)
+		require.NoError(t, err)
+		require.Equal(t, expectedCount, count)
+	}
+
+	// Check team 1 counts
+	assertHostCount("CVE-2024-1000", 1, 1, false)
+	assertHostCount("CVE-2024-2000", 1, 10, false)
+
+	// Check team 2 counts
+	assertHostCount("CVE-2024-1000", 2, 2, false)
+	assertHostCount("CVE-2024-2000", 2, 20, false)
+
+	// Check global counts
+	assertHostCount("CVE-2024-1000", 0, 4, true)
+	assertHostCount("CVE-2024-2000", 0, 40, true)
+
+	// Check no team counts
+	assertHostCount("CVE-2024-1000", 0, 1, false)
+	assertHostCount("CVE-2024-2000", 0, 10, false)
```

Comment:
Can you use different numbers here? Team 1 counts are the same as Team 0 counts -- ideally, all counts should be unique. Maybe CVE 1000 should have a count of 0 (a good edge case), and CVE 2000 a count of 11?

Also, what happens if the counts are off somehow and no team counts ends up negative?

Reply:
good stuff, i updated tests and added `GREATEST()` to eliminate negative counts